cl/search/tests.py

```diff
@@ -3670,3 +3670,91 @@ def test_oa_results_relevance_ordering_elastic(self) -> None:
             < r.content.decode().index("Jose"),
             msg="'Hong Liu Yang' should come BEFORE 'Hong Liu Lorem' and 'Jose' when order_by relevance.",
         )
+
+    def test_oa_results_api_fields_es(self) -> None:
+        search_params = {
+            "type": SEARCH_TYPES.ORAL_ARGUMENT,
+            "q": "Hong Liu Lorem v. Lynch-Loretta E.",
+        }
+
+        # API
+        r = self.client.get(
+            reverse("search-list", kwargs={"version": "v3"}), search_params
+        )
+        actual = self.get_results_count(r)
+        expected = 1
+        self.assertEqual(actual, expected)
+
+        keys_to_check = [
+            "absolute_url",
+            "caseName",
+            "court",
+            "court_citation_string",
+            "court_exact",
+            "court_id",
+            "dateArgued",
+            "dateReargued",
+            "dateReargumentDenied",
+            "docketNumber",
+            "docket_id",
+            "download_url",
+            "duration",
+            "file_size_mp3",
+            "id",
+            "judge",
+            "local_path",
+            "panel_ids",
+            "snippet",
+            "source",
+            "timestamp",
+        ]
+        keys_count = len(r.data["results"][0])
+        self.assertEqual(keys_count, 21)
+        for key in keys_to_check:
+            if key in r.data["results"][0]:
+                key_is_present = True
+            else:
+                key_is_present = None
+            self.assertIsNotNone(
+                key_is_present,
+                msg=f"Key {key} not found in the result object.",
+            )
+
+    def test_oa_results_api_pagination(self) -> None:
+        with transaction.atomic():
```

Comment:
I can never remember, but isn't each test run in a transaction anyway?

Reply:
Yes, test run in a transaction. The issue at hand is that every time a new Audio object is created, it's automatically indexed in ES due to Django ES signals. This test creates multiple objects, causing them to be added to the index, which in turn causes other tests to fail due to content mismatches.

So I wanted to do something like:

```
    with transaction.atomic():
        # Create new Audio objects
        # ES index is updated.
        # Do testing
        
   # Outside transaction block,  DB was not modified but the index was.
   # Recreate index based on current objects in DB.
```

However, it appears this isn't necessary. We can proceed with the test as usual, and at the end, remove only the Audio objects that were created during this test. The deletion action will trigger an ES update (via `post_delete` signal), removing only these objects from the index. This is changed.